### PR TITLE
Relax and document thread-safety for JSON parsing

### PR DESCRIPTION
**Pull request by eile**
Stumbled upon this when creating independent events in parallel: When
their types are not registered, the (unneeded) schema parsing in the
event ctor called in parallel unordered_map::operator [] which is not
thread safe.

Moved schema parsing to JSON code and made getSchema() "const"

### CODE REVIEW

**Comment by hernando** [comment]
Typo on commit message "mad"

File: zeq/vocabulary.h [hunk 1]
```diff
@@ -59,7 +61,22 @@ ZEQ_API Event serializeRequest( const uint128_t& eventType );
  * @return an uint128_t to identify the zeq event that should be published.
  */
 ZEQ_API uint128_t deserializeRequest( const Event& event );
+//@}
 
+/**
+ * @name JSON/binary event translation.
+ *
+ * These functions are not thread-safe, that is, calling registerEvent()
```
Comment (tribal-tec) [inline zeq/vocabulary.h]: You call that function in user code? Where and why? Usually, flatbuffers generated calls that on init, so I'm wondering what use case I'm missing.

File: zeq/detail/event.h [hunk 2]
```diff
@@ -25,13 +23,7 @@ class Event
...
     Event( const uint128_t& type_ )
         : type( type_ )
-    {
-        const std::string& schema = vocabulary::detail::getSchema( type );
-
-        // populate parser with schema from type
-        if( !schema.empty() && !parser.Parse( schema.c_str( )))
-            LBTHROW( std::runtime_error( parser.error_ ));
-    }
+    {}
```
Comment (tribal-tec) [inline zeq/detail/event.h]: that change in behavior will affect RESTBridge if I'm not mistaken. Why the change?

**Comment by tribal-tec** [comment]
@chevtche @favreau  Can you check if this ZEQ version would work with RESTBridge?

**Comment by tribal-tec** [comment]
Document exceptions accordingly for Event ctor and (de)serializeJSON functions.
+1 then.
